From: Jan-Niklas Burfeind <git@aiyionpri.me>
Date: Sun, 26 Mar 2023 22:50:42 +0200
Subject: ipq40xx: add reset button for Google WiFi (Gale)

Add the external reset button for use with OpenWrt.

Co-authored-by: Brian Norris <computersforpeace@gmail.com>
Signed-off-by: Jan-Niklas Burfeind <git@aiyionpri.me>
Reviewed-by: Brian Norris <computersforpeace@gmail.com>
Tested-by: Brian Norris <computersforpeace@gmail.com>
Signed-off-by: David Bauer <mail@david-bauer.net>
(cherry picked from commit 791550b94fc791866e486e52d3c4d1d0378966ff)

diff --git a/target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-wifi.dts b/target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-wifi.dts
index 9448e5145e29c39d4f7bfc2b94a745b2ccc1cf2d..c48d6a75b391886840d1889ff0d3b091898e4996 100644
--- a/target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-wifi.dts
+++ b/target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-wifi.dts
@@ -39,16 +39,29 @@
 			status = "okay";
 		};
 	};
+
+	keys {
+		compatible = "gpio-keys";
+		pinctrl-0 = <&fw_pinmux>;
+		pinctrl-names = "default";
+
+		reset {
+			label = "reset";
+			gpios = <&tlmm 57 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
 };
 
 &tlmm {
-	fw_pinmux {
+	fw_pinmux: fw_pinmux {
 		wp {
 			pins = "gpio53";
 			output-low;
 		};
 		recovery {
 			pins = "gpio57";
+			function = "gpio";
 			bias-none;
 		};
 		developer {
